### DISCUSSION TITLE
Is there a way to set a default task trigger rule for a DAG?

### DISCUSSION THREAD

**Discussion by p-hebert** [apache/airflow#23098] [Q&A] [answered]
I created an operator that can be skipped based on a list of tags passed in the DagRun conf.
The issue I have is that all other non-skippable operators by default have the default trigger_rule `ALL_SUCCESS`, which I would like to change to `NONE_FAILED`. Is there a way I can do that directly when initiating a Dag or a TaskGroup object? This would allow me to make cleaner code by abstracting the trigger_rule override to the Dag/TaskGroup level instead of doing it on every task.

**Comment by josh-fell** [comment, marked as answer]
You can use `default_args` at either the DAG or TaskGroup level (or both actually if you need to) to apply a single `trigger_rule` value to every task.

Given this example:
```python
from pendulum import datetime

from airflow.decorators import dag, task
from airflow.utils.trigger_rule import TriggerRule


@dag(
    start_date=datetime(2022, 1, 1),
    schedule_interval=None,
    default_args={"trigger_rule": TriggerRule.NONE_FAILED},
)
def default_trigger_rules():
    @task
    def func():
        ...

    @task
    def other_func():
        ...

    func() >> other_func()


_ = default_trigger_rules()
```
Inspecting DAGs in the DagBag and check the tasks `trigger_rule` values shows that "none_failed" is applied to both tasks in the DAG.
<img width="1139" alt="image" src="https://user-images.githubusercontent.com/48934154/164133491-d74abaed-7986-4752-96d3-a8d97df67b0f.png">

**Reply by p-hebert** [reply]
Lovely, thanks @josh-fell ! 🤩
